app/actions/websocket/threads.ts

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {processReceivedThreads, processUpdateTeamThreadsAsRead, processUpdateThreadFollow, processUpdateThreadRead} from '@actions/local/thread';
+import DatabaseManager from '@database/manager';
+
+export async function handleThreadUpdatedEvent(serverUrl: string, msg: WebSocketMessage): Promise<void> {
+    try {
+        const thread = JSON.parse(msg.data.thread) as Thread;
+        thread.is_following = true; // Mark as following
+        processReceivedThreads(serverUrl, [thread]);
+    } catch (error) {
+        // Do nothing
+    }
+}
+
+export async function handleThreadReadChangedEvent(serverUrl: string, msg: WebSocketMessage): Promise<void> {
+    const database = DatabaseManager.serverDatabases[serverUrl];
+    if (!database) {
+        return;
+    }
+
+    try {
+        const operator = database?.operator;
+        if (operator) {
+            const {thread_id, timestamp, unread_mentions, unread_replies} = msg.data as {
```

Comment:
Extract to type? (Could be in this file, just before this function, just to make the code a bit more readable)